rslib/src/scheduler/fsrs/weights.rs

```diff
@@ -216,7 +248,8 @@ pub(crate) fn single_card_revlog_to_items(
         let manually_rescheduled =
             entry.review_kind == RevlogReviewKind::Manual || entry.button_chosen == 0;
         let cram = entry.review_kind == RevlogReviewKind::Filtered && entry.ease_factor == 0;
-        if manually_rescheduled || cram {
+        let before_date = entry.cid.0 < review_revlogs_before.into();
```

Comment:
`cid` is the timestamp of created time of the card. It is unequal to the first review time.

Reply:
Well spotted!

Reply:
I find out a new problem. When training, we ignore cards that don't have any learning steps. What if the learning steps are removed in line 251?